README.md

```diff
@@ -14,9 +14,21 @@ Online documentation is available at https://mozilla.github.io/mozanalysis/
 pip install mozanalysis
 ```
 
+## Local Installation
+```bash
+# Create and activate a python virtual environment.
+python3 -m venv venv/
+source venv/bin/activate
+pip install -r requirements.txt
+pip install -e .
+```
+
 ## Development
+After installing the dependencies as described above, install the package and additional development dependencies with `pip install -e ".[dev]"`
 Linting and Formatting are done with Ruff.
 
+When adding new dependencies, add them to the `pyproject.toml` `dependencies` list.  Then update the `requirements.txt` file using the script `script/update_deps`.  [`pip-compile` is called](https://pypi.org/project/pip-tools/) in this script, which uses the specified dependencies to create the `requirements.txt` file.
```

Comment:
Oh one other thing -- consider adding a note to re-run `pip install -r requirements.txt` after regenerating it. Easy to forget (at least for me).

Reply:
Good call, pushed an update